core/src/action.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+  Copyright:    © 2023 SIL International.
+  Description:  Implementation of the action API functions using internal
+                data structures and functions.
+  Create Date:  23 Oct 2023
+  Authors:      Marc Durdin (MCD)
+  History:      23 Oct 2023 - MCD - Initial implementation from #9720
+*/
+#include <cassert>
+#include <algorithm>
+#include <sstream>
+
+#include <keyman/keyman_core_api.h>
+
+#include "action.hpp"
+#include "state.hpp"
+#include "option.hpp"
+
+km_core_actions * km::kbp::action_items_to_actions(
```

Comment:
```suggestion
km_core_actions * km::kbp::action_item_list_to_actions_object(
```

Too bad we can't use markdown in the function name here! 

clarifying name

Reply:
thought about 'flatten' in the name also

Reply:
this is internal, not an api, so I am less worried about it, will accept this suggestion